compute_endpoint/tests/unit/test_working_dir.py

```diff
@@ -0,0 +1,143 @@
+import os
+import shutil
+import uuid
+
+import pytest
+from globus_compute_common import messagepack
+from globus_compute_endpoint.engines.globus_compute import GlobusComputeEngine
+from globus_compute_endpoint.engines.helper import execute_task
+from globus_compute_sdk.serialize import ComputeSerializer
+from tests.utils import ez_pack_function
+
+
+@pytest.fixture()
+def reset_cwd():
+    cwd = os.getcwd()
+    yield
+    os.chdir(cwd)
+
+
+def get_cwd():
+    import os
+
+    return os.getcwd()
+
+
+def test_working_dir_default(tmp_path):
+
+    gce = GlobusComputeEngine()
+
+    gce.start(endpoint_id=uuid.uuid4(), run_dir=tmp_path.name)
+    assert gce.working_dir == tmp_path.name
+    gce.shutdown()
+    shutil.rmtree(tmp_path.name)
+
```

Comment:
As discussed privately, I think that `shutil.rmtree` hides the larger fact that this test isn't testing exactly what needs to be tested.  In particular, even if later removed, the directories should not be created in the CWD in the first place.  Poke me either here or in Slack if it would be helpful to work through an approach.

Reply:
I believe this should be fixed now. @khk-globus Could you take a look please?